lib/cubits/tickets/tickets_cubit.dart

```diff
@@ -13,8 +13,12 @@ class TicketsCubit extends Cubit<TicketsState> {
   TicketsCubit(this._ticketRepository) : super(const TicketsLoading());
 
   Future<void> getTickets() async {
+    // Only fetch tickets from backend if they are not loaded already
+    if (state is TicketsLoaded) {
```

Comment:
I'm curious, when is this check relevant?

Reply:
Without that check, I was seeing that the tickets got refreshed whenever I navigated to another page in the app from the bottom bar. That seemed wasteful, hence the check. You can try removing it locally and seeing for yourself if my description does not make sense.

Reply:
Seems like TicketsLoading is emitting when navigating to another tab, which is a bug that should be fixed

Reply:
Seems like getTickets (also fetchReceipts) are called on each bottom bar navigation.

Following code in home_page.dart should be refactored:

```dart
        BlocProvider.value(
          value: sl.get<TicketsCubit>()..getTickets(),
        ),
        BlocProvider.value(
          value: sl.get<ReceiptCubit>()..fetchReceipts(),
        ),
```

I can look into it.

Reply:
#247 should fix it

Reply:
https://github.com/AnalogIO/coffeecard_app/pull/247 merged, fixed